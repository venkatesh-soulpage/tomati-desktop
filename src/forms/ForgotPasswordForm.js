import React from "react";
//redux
import { connect } from "react-redux";
import { forgetPassword, receiveForgotPasswordError } from "_actions/auth";
// React router
import { Link } from "react-router-dom";
//react bootstrap
import Form from "react-bootstrap/Form";
import Button from "react-bootstrap/Button";
import Alert from "react-bootstrap/Alert";

function ForgotPasswordForm(props) {
  const [values, setValues] = React.useState({
    email: ""
  });

  function onFormSubmit(event) {
    event.preventDefault();
    var postData = {
      email: values.email
    };
    props.dispatch(forgetPassword(postData));
  }

  return (
    <Form onSubmit={onFormSubmit}>
      <Form.Group>
        {props.auth.forgotPasswordError.status === "ERROR" && (
          <Alert
            variant="danger"
            onClose={() => {
              receiveForgotPasswordError({});
            }}
            dismissible
          >
            Email Doesn't exist please try again with another.
          </Alert>
        )}
      </Form.Group>
      <h3>Reset Your Password</h3>
      <Form.Group>
        <Form.Label>
          Please enter your email address below to receive a link to reset your
          password.
        </Form.Label>
        <Form.Control
          type="email"
          value={values.email}
          onChange={event => {
            setValues({ ...values, email: event.target.value });
          }}
          placeholder="Email"
          required
        ></Form.Control>
      </Form.Group>
      <Form.Group>
        <Button type="submit" variant="primary">
          Submit
        </Button>
      </Form.Group>
      <div className="text-muted">
        <Link className="btn btn-link" to="/login">
          Back to Login
        </Link>
      </div>
    </Form>
  );
}

function mapStateToProps(state) {
  return { auth: state.auth };
}
export default connect(mapStateToProps)(ForgotPasswordForm);
