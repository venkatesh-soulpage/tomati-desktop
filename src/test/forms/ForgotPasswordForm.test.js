import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import ForgotPasswordForm from "forms/ForgotPasswordForm";
import { Router } from "react-router-dom";
import history from "utils/history";
import { Provider } from "react-redux";
import configureStore from "store";

describe("ForgotPasswordForm page", () => {
  it("before entering values", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ForgotPasswordForm />
        </Router>
      </Provider>,
      {}
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("after entering values", () => {
    const store = configureStore();
    const { asFragment } = render(
      <Provider store={store}>
        <Router history={history}>
          <ForgotPasswordForm />
        </Router>
      </Provider>,
      {}
    );
    let inputNode_email = screen.getByPlaceholderText("Email");
    fireEvent.change(inputNode_email, {
      target: {
        value: "tomati@gmail.com",
      },
    });
    expect(inputNode_email.value).toBe("tomati@gmail.com");

    expect(asFragment()).toMatchSnapshot();
  });
});
