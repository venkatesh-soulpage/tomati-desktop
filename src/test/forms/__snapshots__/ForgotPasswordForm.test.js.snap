// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPasswordForm page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-light container-fluid d-flex justify-content-center align-items-center h-75"
  >
    <div
      class="container "
    >
      <h3
        class="text-center"
      >
        Forgot Password
      </h3>
      <div
        class="mt-5 card"
        style="border-radius: 15px; margin: auto;"
      >
        <form
          class="p-5"
        >
          <small
            class="text-secondary d-flex justify-content-center p-3"
          >
            Please enter the email address associated with this account
          </small>
          <div
            class="form-group"
          >
            <div
              class="input-group"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="bg-white border-right-0 input-group-text"
                >
                  <img
                    alt="mail"
                    class="img-fluid"
                    height="20"
                    src="Mail.svg"
                    width="20"
                  />
                </span>
              </div>
              <input
                class="border-left-0 form-control"
                placeholder="Email"
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
          <div
            class="d-flex align-items-center"
          >
            <a
              class="red-link"
              href="/"
            >
              Back to Login
            </a>
            <button
              class="ml-auto  btn btn-primary"
              style="border-radius: 20px;"
              type="submit"
            >
              Retrieve
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
